### PR TITLE
Updated post functions and added necessary models for database

### PR DESCRIPTION
**Pull request by hazerb**
Updated post functions according to new design, also improved design of database, added necessary models and made relationships between some models. Please review.

### CODE REVIEW

File: backend/model/db.js [hunk 1]
```diff
@@ -1,14 +1,31 @@
...
+const Project = ProjectModel(sequelize, Sequelize)
+const UserProject = UserProjectModel(sequelize,Sequelize)
+const ProjectTag = ProjectTagModel(sequelize, Sequelize)
+const ProjectCollaborator = ProjectCollaboratorModel(sequelize, Sequelize)
+const ProjectFile = ProjectFileModel(sequelize, Sequelize)
+
+Project.hasMany(ProjectTag, {foreignKey : 'project_id' , onDelete: 'CASCADE' })
+Project.hasMany(ProjectCollaborator, {foreignKey : 'project_id' , onDelete: 'CASCADE' })
+UserProject.belongsTo(Project, {foreignKey : 'project_id', onDelete: 'CASCADE'})
+ProjectCollaborator.belongsTo(User, {foreignKey: 'user_id'})
```
Comment (FarukOzderim) [inline backend/model/db.js:25]: I think we might also have "onDelete: 'CASCADE'" on ProjectCollaborator user_id.
The design choice should be decided, will the program be able to delete users from DB and will associated Collaboration deleted from ProjectCollaborator table?
```suggestion
ProjectCollaborator.belongsTo(User, {foreignKey: 'user_id'})
```

**Review by FarukOzderim — CHANGES_REQUESTED** [review]

File: backend/model/db.js (hunk 1 above)
Comment (FarukOzderim) [reply backend/model/db.js:25]: We talked over this issue, currently it is not necessary.

Comment (hazerb) [reply backend/model/db.js:25]: i think its not necessary for the first milestone, we could decide it later. It functions well right now

**Comment by isiktashamza** [comment]
I reviewed the code and  I think it is well written and fits into the architecture

**Pull request merged**